### PR TITLE
Tdimhcsleumas/user invtes

### PR DESCRIPTION
**Pull request by tdimhcsleumas**
This pr provides facilities for inviting existing users in our app, creating TeamInvite objects. These objects are then used to create member objects when a user accepts them. There are a few endpoints that are added as a result of this.

From a team admin's perspective,
- POST `/api/teams/{team_pk}/invites` create a TeamInvite entry, which can be later accepted by a user.
- GET `/api/teams/{team_pk}/invites` lists all existing TeamInvite entries for the team.
- DELETE `/api/teams/{team_pk}/invites/{pk}` simply deletes the TeamInvite entry specified by `<pk>`.

I don't think we would need to support updating, since each entry is simply a <team, user> pair. If an admin mistakenly invite the wrong user, simply retract the invite by deleting it.

The TeamInvite object is a <team, user> pair. Since we identify user's by email, the json for both creating is just:
```
{'user_email': <email>}
```

From a user's perspective,
- GET `/users/invites` will list all teams that a user has invites for.
- PUT `/users/invites/{id}` will accept the same json that creating a team does (just the name of the team, since these are unique in our case), delete the invite entry associated with that team, and create a member record for that team.
- DELETE relies on the destroy model mixing and is deliberately untested to avoid contention with jira tasks (since I think this was Troy's task originally).

The TeamInvite gets serialized to users as follows:
```
{
   'id': 12,
   'team': { team json }
}
```

Since we're adopting this creation scheme, endpoints to create member directly are no longer supported.

I went through all our files with more than one class and gave each class its own file. This includes expanding our test scripts, view files, model files, and serializer files. I think this should make the codebase a little bit more maintainable.

### CODE REVIEW

File: api/models/member.py [hunk 1]
```diff
@@ -3,21 +3,24 @@
 from django.utils.translation import gettext_lazy as _
 from django.contrib.auth import get_user_model
 from hashlib import md5
-import uuid
+from django.db.models.signals import pre_delete
+from django.dispatch import receiver
```
Comment (agavgavi) [inline api/models/member.py:7]: These are not used.

File: api/models/tag.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if not title:
+            raise ValueError("missing title")
+
+        if not team:
+            raise ValueError("missing team")
+
+        title_id = "{}".format(title)
+        team_id = "{}".format(team.id)
+        self.team_title_hash = md5(title_id.encode()).hexdigest() + md5(
+            team_id.encode()).hexdigest()
```
Comment (agavgavi) [inline api/models/tag.py:52]: this can be replaced with your new function right?

**Review by agavgavi — APPROVED** [review]
This is great work! I love your implementation of arbitrary users/teams for testing, and the refactor of how everything is laid out in folders are super useful.

The only comments I had were unused imports or functions, but besides that it looks great.

File: api/models/member.py (hunk 1 above)
Comment (tdimhcsleumas) [reply api/models/member.py]: Fixed!

File: api/models/tag.py (hunk 2 above)
Comment (tdimhcsleumas) [reply api/models/tag.py]: Updated!

**Pull request merged**